IPython/core/interactiveshell.py

```diff
@@ -482,7 +487,9 @@ def __init__(self, ipython_dir=None, profile_dir=None,
 
         # This is where traits with a config_key argument are updated
         # from the values on config.
+        self._preinit = True
```

Comment:
yep, avoid deadloc durring the call of super() which trigger _style_changed, which need init_syntax_highlighting to have been called, which need super() to have been called, which need init_syntax_highlighting to have been called, which need... super()